packages/components/src/test/tree/ReportingTreeNodeLoader.test.ts

```diff
@@ -0,0 +1,120 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { expect } from "chai";
+import { of, Observable as RxjsObservable, take } from "rxjs";
+import sinon from "sinon";
+import { ITreeNodeLoader, Observable, PagedTreeNodeLoader, TreeDataProvider, TreeModelNode, TreeNodeItem, TreeNodeLoadResult } from "@itwin/components-react";
+import { ReportingTreeNodeLoader } from "../../presentation-components/tree/ReportingTreeNodeLoader";
+import { waitFor } from "../TestUtils";
+
+describe("ReportingTreeNodeLoader", () => {
+  let reportingNodeLoader: ReportingTreeNodeLoader<TreeDataProvider>;
+  const reportStub = sinon.stub<[{ node: string | undefined; duration: number }], void>();
+  const nodeLoaderStub = {
+    loadNode: sinon.stub<Parameters<ITreeNodeLoader["loadNode"]>, ReturnType<ITreeNodeLoader["loadNode"]>>(),
+    modelSource: {},
+    dataProvider: {},
+  };
+
+  beforeEach(() => {
+    nodeLoaderStub.loadNode.returns(
+      of(
+        { loadedNodes: [{ id: "node 1" }] } as TreeNodeLoadResult,
+        { loadedNodes: [{ id: "node 2" }] } as TreeNodeLoadResult,
+        { loadedNodes: [{ id: "node 3" }] } as TreeNodeLoadResult,
+      ),
+    );
+    reportingNodeLoader = new ReportingTreeNodeLoader(nodeLoaderStub as unknown as PagedTreeNodeLoader<TreeDataProvider>, reportStub);
+  });
+
+  afterEach(() => {
+    nodeLoaderStub.loadNode.reset();
+    reportStub.reset();
+  });
+
+  function toRxjsObservable<T>(source: Observable<T>): RxjsObservable<T> {
+    return new RxjsObservable((subscriber) => source.subscribe(subscriber));
+  }
+
+  describe("loadNode", () => {
+    it("reports node loading duration", async () => {
+      const performanceStub = sinon.stub(performance, "now");
+      performanceStub.onCall(0).returns(0);
+      performanceStub.onCall(1).returns(300);
+
+      const observable = reportingNodeLoader.loadNode({ id: "id" } as TreeModelNode, 0);
+      let loadedNodes: TreeNodeItem[] = [];
```

Comment:
I think there is no need to track loaded nodes. Also maybe it would be simpler to use `Subject` instead of `of` to mock `loadNode`. Eg:

```ts
const lodeNodeSubject = new Subject<TreeNodeLoadResult>();
nodeLoaderStub.loadNode.returns(loadNodeSubject);

reportingNodeLoader.loadNode({ id: "id" } as TreeModelNode, 0).subscribe();
nodeLoadSubject.complete();

await waitFor(() => {
  expect(reportStub).to.be.calledWithMatch({ node: "id", duration: 300 });
});
```

Same in other tests. `Subject` will allow you to manually control when `loadNode` observable completes/errors.

Reply:
I am aware that tracking the loaded nodes is not necessary for testing whether the load duration is reported, however we should probably also test to see if the `ReportingTreeNodeLoader` returns the same result as the wrapped loader?

Reply:
Ok we can leave nodes tracking but I still think using `Subject` simplifies tests.

Reply:
Switched to `Subject`.